util/ranger/detacher.go

```diff
@@ -133,61 +136,185 @@ func getEqOrInColOffset(expr expression.Expression, cols []*expression.Column) i
 	return -1
 }
 
-func extractAccessAndFilterConds(conditions, accessConds, filterConds []expression.Expression,
-	col *expression.Column, length int) ([]expression.Expression, []expression.Expression) {
+func extractAccessAndFilterConds(conditions, accessConds, filterConds []expression.Expression, colName model.CIStr,
```

Comment:
The name of this function is not very good. This function only serves for a particular column